js/src/filesystem.ts

```diff
@@ -0,0 +1,128 @@
+/******************************************************************************
```

Comment:
Can you please split the lint changes and the file moves into two separate commits? Otherwise, git thinks that this is a new file (same problem in a couple of other places too). This breaks various history viewer/search tools, and sometimes makes rebases go crazy

Reply:
ive split them into two commits

Reply:
thanks!